sdks/python/apache_beam/io/filesystem.py

```diff
@@ -330,6 +334,10 @@ def _get_compression_type(path, compression_type):
                       'was %s' % type(compression_type))
     return compression_type
 
+  @staticmethod
```

Comment:
How about moving this to CompressedFile class or simply doing isinstance(fileobj, CompressedFile) at places where this is currently called ?

In current form, this gives the false impression that we are checking with the underlying fie-system impl to see if the file is compressed.